### PR TITLE
wy50: add beep

### PR DESCRIPTION
**Pull request by goldnchild**
I was using the wy50 with nabu cp/m and thought I'd try to add the beeper.  

I don't know the proper frequency of the wy50 beeper so I made a port_changed_member so you could adjust the beeper clock from the slider controls menu.

### CODE REVIEW

**Comment by happppp** [comment]
2048Hz

But even if the freq was unknown, you could just add a comment saying it's guessed. Instead of a slider configuration.
m_speaker class object is not needed btw.

File: src/mame/wyse/wy50.cpp [hunk 1]
```diff
@@ -36,6 +36,8 @@
 #include "wy50kb.h"
 #include "video/scn2674.h"
 #include "screen.h"
+#include "speaker.h"
+#include "sound/beep.h"
```
Comment (cuavas) [inline src/mame/wyse/wy50.cpp:40]: These should be kept organised – see here: https://docs.mamedev.org/contributing/cxx.html#structural-organization

File: src/mame/wyse/wy50.cpp [hunk 2]
```diff
@@ -82,6 +88,8 @@ class wy50_state : public driver_device
 	required_device<er1400_device> m_earom;
 	required_device<scn2672_device> m_pvtc;
 	required_device<scn2661b_device> m_sio;
+	required_device<speaker_device> m_speaker;
+	required_device<beep_device> m_beep;
```
Comment (cuavas) [inline src/mame/wyse/wy50.cpp:92]: `m_speaker` is unnecessary.

File: src/mame/wyse/wy50.cpp [hunk 3]
```diff
@@ -280,8 +290,15 @@ void wy50_state::row_buffer_map(address_map &map)
...
 static INPUT_PORTS_START(wy50)
+	PORT_START("BEEPCLOCK")
+	PORT_BIT(0xffff, 830, IPT_ADJUSTER) PORT_MINMAX(0x000, 0x1000) PORT_CHANGED_MEMBER(DEVICE_SELF, wy50_state, adjust_beep, 0) PORT_NAME("Adjust Beep Frequency")
 INPUT_PORTS_END
 
+INPUT_CHANGED_MEMBER( wy50_state::adjust_beep )
+{
+	m_beep->set_clock(newval);
+}
+
```
Comment (cuavas) [inline src/mame/wyse/wy50.cpp:301]: This is unnecessary complexity.

File: src/mame/wyse/wy50.cpp [hunk 4]
```diff
@@ -320,6 +337,9 @@ void wy50_state::wy50(machine_config &config)
 	modem.rxd_handler().set(m_sio, FUNC(scn2661b_device::rxd_w));
 	modem.cts_handler().set(m_sio, FUNC(scn2661b_device::cts_w));
 	modem.dcd_handler().set(m_sio, FUNC(scn2661b_device::dcd_w));
+
+	SPEAKER(config, "speaker").front_center();
+	BEEP(config, m_beep, 950).add_route(ALL_OUTPUTS, "speaker", 0.10);  // Star Micronics QMB06 PZT Buzzer
```
Comment (cuavas) [inline src/mame/wyse/wy50.cpp:342]: You can find specifications for this device trivially: https://www.electronicsurplus.com/star-qmb06-audio-alarms-and-buzzers-pzt-sound-transducer-buzzer

**Review by cuavas — CHANGES_REQUESTED** [review]

File: src/mame/wyse/wy50.cpp (hunk 4 above)
Comment (goldnchild) [reply src/mame/wyse/wy50.cpp]: For some reason the frequency 2048 sounds too high.  The only video I could find on youtube with the beep sounds is this one, right at the beginning of the video and at the 9:32 mark:

https://www.youtube.com/watch?v=xQTr9ZOJkC0
https://youtu.be/xQTr9ZOJkC0?t=570

File: src/mame/wyse/wy50.cpp (hunk 3 above)
Comment (goldnchild) [reply src/mame/wyse/wy50.cpp]: ok

File: src/mame/wyse/wy50.cpp (hunk 4 above)
Comment (happppp) [reply src/mame/wyse/wy50.cpp]: QMB06 datasheet says 2048hz, are you sure that's the buzzer on this WY50/75?
Try 1024hz then, that's more close to what that video sounds like.

Comment (goldnchild) [reply src/mame/wyse/wy50.cpp]: Looking at the maintentance manual:

https://doc.lagout.org/science/0_Computer Science/0_Computer History/old-hardware/wyse/WY-50/Wyse_WY-50_Terminal_Mainenance_198601.pdf

theres a bunch of schematics in the pdf manual of different versions, p213 for example has the STAR QMB06 shown in a circuit that has a bunch of resistors and capacitors and 74LS14s and diodes along with an LED.  I wonder if that affects the sound of the buzzer.

Comment (goldnchild) [reply src/mame/wyse/wy50.cpp]: <a href="https://imgur.com/MuxYEQY"></a>

Comment (goldnchild) [reply src/mame/wyse/wy50.cpp]: Looking at the datasheet it looks like there's a bunch of resonant frequency peaks, if I read it right, one about 660hz too.

<a href="https://imgur.com/0YhREwV"></a>

        SPEAKER(config, "speaker").front_center();
        BEEP(config, m_beep,  2048).add_route(ALL_OUTPUTS, "speaker", 0.10);  // Star Micronics QMB06 PZT Buzzer
        BEEP(config, m_beep2,  680).add_route(ALL_OUTPUTS, "speaker", 0.05);  // Star Micronics QMB06 PZT Buzzer
        BEEP(config, m_beep3, 4500).add_route(ALL_OUTPUTS, "speaker", 0.05);  // Star Micronics QMB06 PZT Buzzer
        BEEP(config, m_beep4, 8800).add_route(ALL_OUTPUTS, "speaker", 0.025);  // Star Micronics QMB06 PZT Buzzer

        m_beep->set_state(BIT(data, 5));
        m_beep2->set_state(BIT(data, 5));
        m_beep3->set_state(BIT(data, 5));
        m_beep4->set_state(BIT(data, 5));


If I make multiple beep devices for each one of the peaks, it has a different sound, it makes it sound lower than 2048 hz.

Maybe 4 different beepers is overkill, but having 2 (at 2048hz and 680hz) sounds better than just the single beeper at 2048hz.

Comment (happppp) [reply src/mame/wyse/wy50.cpp]: I see, that looks like it's an LC filter. See the coil, and then the cap to GND.
Multiple beepers? nah, I wouldn't do that.

SPEAKER(config, "speaker").front_center();
// Star Micronics QMB06 PZT Buzzer (2048Hz peak) + LC filter, output frequency is approximated here
BEEP(config, m_beep, 1000).add_route(ALL_OUTPUTS, "speaker", 0.10);

Comment (goldnchild) [reply src/mame/wyse/wy50.cpp]: ok, I put the single beeper back with your LC comment and 1000 hz beeper.

SPEAKER(config, "speaker").front_center();
// Star Micronics QMB06 PZT Buzzer (2048Hz peak) + LC filter, output frequency is approximated here
BEEP(config, m_beep, 1000).add_route(ALL_OUTPUTS, "speaker", 0

**Pull request merged**